common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfTestBase.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.perf.test.core;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Performance Test Base class.
+ * @param <TOptions> the options used to configure the options class.
+ */
+public abstract class PerfTestBase<TOptions extends PerfStressOptions> {
+    protected final TOptions options;
+    protected long lastCompletionNanoTime;
+    protected long completedOperations;
```

Comment:
I think `protected long completedOperations` should be moved to `ApiPerfTestBase`, since it's not used by `EventPerfTest` due to thread-safety.